sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ParametersCorrectOrder.cs

```diff
@@ -124,7 +124,7 @@ protected sealed override void Initialize(SonarAnalysisContext context)
 
                 if (identifierArgument is PositionalIdentifierArgument positional &&
                     (parameter.IsParams ||
-                     !identifierNames.Contains(parameterName) ||
+                     !IdentifierWithSameNameAndTypeExists(parameter) ||

```

Comment:
This is probably the most expensive check of the four being made - might be better to put it last so it only runs when necessary.